src/controlnet_aux/processor.py

```diff
@@ -20,84 +20,117 @@
+    'openpose_full': {'resize': False, 'hand_and_face': True},
+    'scribble_pidinet': {'resize': False, 'safe': False, 'scribble': True},
+    'softedge_pidinet': {'resize': False, 'safe': False, 'scribble': False},
+    'scribble_pidsafe': {'resize': False, 'safe': True, 'scribble': True},
+    'softedge_pidsafe': {'resize': False, 'safe': True, 'scribble': False},
+    'normal_bae': {'resize': False},
+    'lineart_realistic': {'resize': False, 'coarse': False},
     'lineart_coarse': {'resize': False, 'coarse': True},
     'lineart_anime': {'resize': False},
     'canny': {'resize': False},
-    'content_shuffle': {'resize': False},
-    'zoe': {'resize': False},
+    'shuffle': {'resize': False},
+    'depth_zoe': {'resize': False},
     'mediapipe_face': {'resize': False},
 }
 
+CHOICES = f"Choices for the processor are {list(MODELS.keys())}"
+
 
 class Processor:
-    def __init__(self, processor_id: str) -> 'Processor':
+    def __init__(self, processor_id: str, params: Optional[Dict] = None) -> None:
         """Processor that can be used to process images with controlnet aux processors
 
         Args:
-            processor_id (str): processor name
-
-        Returns:
-            Processor: Processor object
+            processor_id (str): processor name, options are 'hed, midas, mlsd, openpose,
+                                pidinet, normalbae, lineart, lineart_coarse, lineart_anime,
+                                canny, content_shuffle, zoe, mediapipe_face
+            params (Optional[Dict]): parameters for the processor
         """
-        print(f"Loading {processor_id} processor")
+        LOGGER.info("Loading %s".format(processor_id))
         self.processor_id = processor_id
         self.processor = self.load_processor(self.processor_id)
+
+        # load default params
         self.params = MODEL_PARAMS[self.processor_id]
+        # update with user params
+        if params:
+            self.params.update(params)
+
         self.resize = self.params.pop('resize', False)
         if self.resize:
-            # print warning: image will be resized
-            print(f"Warning: {self.processor_id} will resize image to {self.resize}x{self.resize}")
+            LOGGER.warning(f"Warning: {self.processor_id} will resize image to {self.resize}x{self.resize}")
 
-    def load_processor(self, processor_id: str):
+    def load_processor(self, processor_id: str) -> 'Processor':
         """Load controlnet aux processors
 
         Args:
             processor_id (str): processor name
+
+        Returns:
+            Processor: controlnet aux processor
         """
+        if processor_id not in MODELS:
+            raise ValueError(f"Processor {processor_id} not found. {CHOICES}")
+
         processor = MODELS[processor_id]['class']
 
+        # check if the proecssor is a checkpoint model
         if MODELS[processor_id]['checkpoint']:
             processor = processor.from_pretrained("lllyasviel/Annotators")
```

Comment:
If you add SamDetector, this might have conflict because SAM weight is not on lllyasviel/Annotators

Reply:
I will add SAM in the next PR, currently it seems to need a local checkpoint
```
sam = SamDetector.from_pretrained("./weight_path")
```

Reply:
I will add SAM in the next PR

Reply:
I added support for SAM to work with model data that was on HuggingFace in #49, so we can wait for that PR or grab the changes there into a separate one.